From 6cd7e4a3647b6000c03db008e9dc06a7dc52e875 Mon Sep 17 00:00:00 2001
From: Ramprasad N <x0038811@ti.com>
Date: Fri, 21 Feb 2020 12:07:05 +0530
Subject: [PATCH] v4l2src: Check for drm memory support in try_import

In Gstreamer 1.16.1, if buffer pool is from a downstream
element, v4l2 first tries to import.

Check if the memory type is DRM memory.

Upstream-Status: Pending
Signed-off-by: Ramprasad N <x0038811@ti.com>
Signed-off-by: Xulin Sun <xulin.sun@windriver.com>
---
 sys/v4l2/gstv4l2object.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/sys/v4l2/gstv4l2object.c b/sys/v4l2/gstv4l2object.c
index 2d401a7..5143e2c 100644
--- a/sys/v4l2/gstv4l2object.c
+++ b/sys/v4l2/gstv4l2object.c
@@ -4768,7 +4768,9 @@ gst_v4l2_object_try_import (GstV4l2Object * obj, GstBuffer * buffer)
     for (i = 0; i < n_mem; i++) {
       GstMemory *mem = gst_buffer_peek_memory (buffer, i);
 
-      if (!gst_is_dmabuf_memory (mem)) {
+      if (!gst_is_dmabuf_memory (mem) &&
+          !gst_memory_is_type (mem, GST_ALLOCATOR_DMABUF)) {
+
         GST_DEBUG_OBJECT (obj->dbg_obj, "Cannot import non-DMABuf memory.");
         return FALSE;
       }
-- 
1.9.1

